src/tests/nativeaot/GenerateUnmanagedEntryPoints/GenerateUnmanagedEntryPoints.cpp

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifdef TARGET_WINDOWS
+#include "windows.h"
+#else
+#include "dlfcn.h"
+#endif
+#include "stdio.h"
+#include "string.h"
+
+#ifndef TARGET_WINDOWS
+#define __stdcall
+#endif
+
+// typedef for shared lib exported methods
+typedef void(__stdcall *fptr)();
+
+#ifdef TARGET_WINDOWS
+int __cdecl main()
+#else
+int main(int argc, char* argv[])
+#endif
+{
+#ifdef TARGET_WINDOWS
+    HINSTANCE handle = LoadLibrary("GenerateUnmanagedEntryPoints.dll");
+#elif __APPLE__
+    void *handle = dlopen(strcat(argv[0], ".dylib"), RTLD_LAZY);
+#else
+    void *handle = dlopen(strcat(argv[0], ".so"), RTLD_LAZY);
+#endif
+
+    if (!handle)
+        return 1;
+
+#ifdef TARGET_WINDOWS
+    fptr main_assembly_method = (fptr)GetProcAddress(handle, "SharedLibraryAssemblyMethod");
+    fptr ref1_assembly_method = (fptr)GetProcAddress(handle, "ReferencedAssembly1Method");
+    fptr ref2_assembly_method = (fptr)GetProcAddress(handle, "ReferencedAssembly2Method");
+#else
+    fptr main_assembly_method = (fptr)dlsym(handle, "SharedLibraryAssemblyMethod");
```

Comment:
For the [Microsoft.iOS.dll case](https://github.com/dotnet/runtime/issues/65755#issuecomment-1462436429), do we actually want the entrypoints to be public entrypoints of the library or do we just want them to be named so that they can be statically linked to?

Reply:
(It is ok for the test to stay the way it is. I am just trying to understand the delta between what the test does and what the Xamarin framework is going to do.)

Reply:
Xamarin runtime is compiled as both static (for iOS, tvOS) and dynamic (for macOS, Mac Catalyst) library. Because of the latter, [initializing the managed part](https://github.com/xamarin/xamarin-macios/pull/17374/files#diff-0624df3e0063f45dfca227272d634925dbd44273cd7e21bdb08d236549de1b4eR232) has to be done by [dynamically loading](https://github.com/xamarin/xamarin-macios/pull/17374/files#diff-3345e306b7d55d198dd19c639982237ec04443a02430bc7d611dd0d06df67d4fR27) the exposed entry point. 
That being said, we would need to have public entrypoints.